### DISCUSSION TITLE
Handling PUT & PATCH request in Class based Testing

### DISCUSSION THREAD

**Discussion by PARTHIB-DEB** [encode/django-rest-framework#9125] [Question & Answer]
##  How to manage PUT and PATCH request via APItestcase / class based testing using DRF

-[ ] [Testing By APItestcase](https://www.django-rest-framework.org/api-guide/testing/#api-test-cases)

**Comment by PARTHIB-DEB** [comment]
I have been facing problems recently , Provide a solution if possible

**Comment by Ehsan200** [comment]
Hello there!
I don't get it. what is the problem? could you please provide more details?

**Comment by PARTHIB-DEB** [comment]
I am trying to say that for function based testing there is a way of put and patch request testing but for class based views I can't see any put patch request handling code

**Comment by Ehsan200** [comment]
I think I get it! You can just use `patch` or `put` function rather than `post` in the doc example:

```python
# for put
response = self.client.put(url, data, format='json')
# for patch
response = self.client.patch(url, data, format='json')
```

**Comment by PARTHIB-DEB** [comment]
```bash
        payload_patch_data = {
            "cname":"Best Python Tutorial",
        }
               # Testing PATCH request
        url_patch=reverse('course',args={"pk":f"{payload['cid']}"})
        response_patch=self.client.patch(url_patch,payload_patch_data,format='json')
        self.assertEqual(response_patch.status_code,status.HTTP_200_OK)
        print("\t\tPATCH REQUEST SUCCESSFUL")
```

Actual Its saying 400 != 200 plus I want to recheck the patch data by again from gaining data from get request

**Reply by PARTHIB-DEB** [reply]
I have tried before you told the same way and facing problem......If you can identify please let me know
